src/background/adblocker.js

```diff
@@ -167,7 +168,29 @@ export const setup = asyncSetup([
  * Cosmetics injection
  */
 
-async function injectScriptlets(scripts, tabId, frameId) {
+const contentScripts = (() => {
+  const map = new Map();
+  return {
+    set(key, value) {
+      map.set(key, value);
+    },
+    has(key) {
+      return map.has(key);
+    },
+    delete(key) {
+      const contentScript = map.get(key);
```

Comment:
The only difference from "normal" map is this function, which should be part of the `injectScriptlets`.